components/Toaster.tsx

```diff
@@ -0,0 +1,147 @@
+import { useSyncExternalStore } from 'react'
+import { Animated, View } from 'react-native'
+import { AnimatedFlex } from './common/ui/Flex'
+import { Text } from './common/ui/Text'
+
+type Toast = {
+  id: number
+  text: string
+  duration?: number
+  animatedValue?: Animated.Value
+}
+
+type Action =
+  | {
+      type: 'ADD'
+      toast: Toast
+    }
+  | {
+      type: 'REMOVE'
+      id: number
+    }
+
+const TOAST_LIMIT_POLICY = 5
+
+let toastMemory: Toast[] = []
+let listeners: ((toasts: Toast[]) => void)[] = []
+
+const reducer = (state: Toast[], action: Action): Toast[] => {
+  switch (action.type) {
+    case 'ADD':
+      const newState = [...state, action.toast]
+      return newState.slice(0, TOAST_LIMIT_POLICY)
+    case 'REMOVE':
+      return state.filter((toast) => toast.id !== action.id)
+  }
+}
+
+const emitChange = () => {
+  for (const listener of listeners) {
+    listener(toastMemory)
+  }
+}
+
+const dispatch = (action: Action) => {
+  toastMemory = reducer(toastMemory, action)
+  emitChange()
+}
+
+export const toast = {
+  show: ({ text, duration }: Omit<Toast, 'id'>) => {
+    const newToast = {
+      id: Date.now(),
```

Comment:
[Suggestion]

이론적으로 같은 ID값을 가질 가능성이 있을 것 같아요.
더 안전한 ID 생성하는 방법은 어떠신가요?

```suggestion
      id: () => ++toastId,
```